docs/design/net-aware.md

```diff
@@ -0,0 +1,64 @@
+# Net Topology Aware Plugin
+
+## Backgrounds
+
+Usually, a kubernetes cluster has many nodes, and those nodes are in different idc, chassis, and even different switches.
+Data transformations across different idc, chassis and switches has different performance. Some latency sensitive workloads are need to run in same idc, even in same topology devices, such as chassis and switch.
+
+## Motivation
+
+We target to make scheduler net-topology aware so as to achieve the following:
+
+- best effort to schedule same job to same topology devices, such as same idc, chassis or switch.
+
+## Goals
+
+- Support single-key topology configuration, try to schedule job's all tasks to nodes which have same value with that key
+- Support multiple-key topology policies, the key at front get higher score
+
+## Non-Goals
+
+- Not to find the global solutions among nodes with all values of that key
+
+## Design Action
+
+### Pod scheduling process
+
+1. when the first task of a job is allocated to a node, record the node information in the plugin
+2. when scheduling other tasks, a node with same key-value as the target node, get a higher score. Otherwise, get a zero score.
+3. If a node has multiple keys same as the configured list, the first key matching the configured keys has higher score
+
+```go
+nodeOrderFn := func(task *api.TaskInfo, node *api.NodeInfo) (float64, error){
+    ...
+    score := 0
+    weight := np.weight
+    tlabels := tNode.Node.Labels
+    labels := node.Node.Labels
+    lenth := len(np.topologyKeys)
+    for i, key := range np.topologyKeys {
+        if tlabels[key] == labels[key] {
+            score += (lenth - i) //  key with more priority at front of which with less priority
+            break
+        }
+    }
+    return float64(score * weight), nil
+}
+```
+
+### Usage
+
+1. label nodes with some key-value pairs, for example switch=NvLink-A100, rack=rack1,rack2, idc=bj,sh to partition nodes to different topology zones.
+2. add net-topology plugin in scheduler-configuration
+
+```yaml
+- plugins:
+   - name: net-topology
+     arguments:
+       net-topology.keys: rack,switch,idc
```

Comment:
Is there any other topology needs affinity except network?

Reply:
The key should has a format like `failure-domain.beta.kubernetes.io/zone` or `volcano.sh/xxx`, and user can cofigure the key according their cluster's node configuration.

Reply:
Those keys just is an example. Real keys depend on user's real cluster configuration.